src/main.ts

```diff
@@ -182,6 +184,72 @@ export function generateFile(ctx: Context, fileDesc: FileDescriptorProto): [stri
     chunks.push(generateDataLoadersType());
   }
 
+  if (options.outputSchema) {
+    chunks.push(code`
+      export interface ProtoMetadata {
+        fileDescriptor: ${imp('IFileDescriptorProto@protobufjs/ext/descriptor')};
+        references: { [key: string]: any };
+        dependencies?: ProtoMetadata[];
+      }
+    `);
+
+    const outputFileDesc: IFileDescriptorProto = {
+      ...fileDesc,
+      sourceCodeInfo: null,
+    };
+
+    if (fileDesc.sourceCodeInfo) {
+      outputFileDesc.sourceCodeInfo = {
+        location: fileDesc.sourceCodeInfo.location.filter((loc) => loc['leadingComments'] || loc['trailingComments']),
+      } as any;
+    }
+
+    const values = [code`fileDescriptor: ${JSON.stringify(outputFileDesc)} as any`];
```

Comment:
Fwiw I really wanted to remove this `as any`. Just as is (using `JSON.stringify`), it looks like enums are coming out as strings instead of numbers, which doesn't match the type.

Using `FieldDescriptorProto.toObject` fixes that, but then the problem is that `proto3Optional` is a field in the output, but is not something in the protobufjs `IFieldDescriptorProto` yet (although it should be).

So makes sense to keep the `as any` .